app/src/main/java/de/triology/cas/logout/CesServiceLogoutMessageBuilder.java

```diff
@@ -0,0 +1,62 @@
+package de.triology.cas.logout;
+
+import lombok.val;
+import org.apereo.cas.logout.slo.SingleLogoutMessage;
+import org.apereo.cas.logout.slo.SingleLogoutMessageCreator;
+import org.apereo.cas.logout.slo.SingleLogoutRequestContext;
+import org.apereo.cas.services.RegisteredService;
+import org.apereo.cas.services.RegisteredServiceLogoutType;
+import org.apereo.cas.support.oauth.services.OAuthRegisteredService;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+import org.apereo.cas.ticket.UniqueTicketIdGenerator;
+import org.apereo.cas.util.CompressionUtils;
+import org.apereo.cas.util.DefaultUniqueTicketIdGenerator;
+import org.apereo.cas.util.ISOStandardDateFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CesServiceLogoutMessageBuilder implements SingleLogoutMessageCreator {
+
+    /**
+     * The logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(CesServiceLogoutMessageBuilder.class);
+
+    /**
+     * A ticket Id generator.
+     */
+    private static final UniqueTicketIdGenerator GENERATOR = new DefaultUniqueTicketIdGenerator(18);
+
+    @Override
+    public SingleLogoutMessage create(final SingleLogoutRequestContext request) {
+        TicketGrantingTicket ticket = request.getExecutionRequest().getTicketGrantingTicket();
+        RegisteredService service = request.getRegisteredService();
+        String logoutRequest = "";
+        LOGGER.debug("Generate logout request for: [{}] to [{}]", service.getName(), service.getLogoutUrl());
+
+        if (service instanceof OAuthRegisteredService) {
+            for (String childTicket : ticket.getDescendantTickets()) {
+                if (childTicket.startsWith("AT-")) {
+                    // childTicket starting with AT- is the OAuth access token
+                    logoutRequest = String.format("%s", childTicket);
+                }
+            }
+        } else {
+            // default service message creator
+            logoutRequest = String.format("<samlp:LogoutRequest xmlns:samlp=\"urn:oasis:names:tc:SAML:2.0:protocol\" ID=\"%s\" Version=\"2.0\" "
```

Comment:
This is copied from `DefaultSingleLogoutMessageCreator` right? I would suggest `CesServiceLogoutMessageBuilder` should inherit from `DefaultSingleLogoutMessageCreator` so we could call the parent method here. An acceptable alternative would be to add at least a link here so we can profit from future improvements.

Reply:
Yeah I tried at first but IntelliJ does not seem to find the class.

Reply:
Ahh, I missed another dependency. Now I could extend the Default class